src/webgpu/api/validation/createRenderPipeline.spec.ts

```diff
@@ -442,9 +442,13 @@ TODO: review`
       fragmentShaderCode: t.getFragmentShaderCode(sampleType, componentCount),
     });
 
+    const sampleTypeSuccess =
+      info.sampleType === 'float' || info.sampleType === 'unfilterable-float'
+        ? sampleType === 'float'
+        : info.sampleType === sampleType;
+
```

Comment:
hmm, I feel like success shouldn't be based, on "sample type" since what we're actually checking here is if the pipeline output type is compatible with the format. I don't think we have such a concept yet, and the spec doesn't clearly define "compatible with", so for lack of something better, this LGTM. Maybe we can add a MAINTAINENCE_TODO about improving the condition.

> If descriptor.[entryPoint](https://gpuweb.github.io/gpuweb/#dom-gpuprogrammablestage-entrypoint) has a [pipeline output](https://gpuweb.github.io/gpuweb/wgsl/#pipeline-output) value with [location](https://gpuweb.github.io/gpuweb/wgsl/#input-output-locations) attribute equal to the index of the colorState in the descriptor.[targets](https://gpuweb.github.io/gpuweb/#dom-gpufragmentstate-targets) list:
 >  - The [pipeline output](https://gpuweb.github.io/gpuweb/wgsl/#pipeline-output) type must be compatible with colorState.[format](https://gpuweb.github.io/gpuweb/#dom-gpucolortargetstate-format).

Reply:
There is a known issue in WebGPU SPEC about output format compatibility in ["validating GPURenderPipelineDescriptor(descriptor, device) Arguments"](validating GPURenderPipelineDescriptor(descriptor, device) Arguments):
> ISSUE 50: define what "compatible" means for render target formats.

I think we should resolve this now.

Reply:
I've added a TODO in the descriptor of the test for this issue. PTAL, thanks!

Reply:
Thanks, LGTM!